### ISSUE TITLE
20230712 - day 26

### ISSUE THREAD

**Issue by tjkpolisher** [tjkpolisher/Bootcamp_TIL#26]
# 들어가기 전에
  오늘 심한 감기몸살로 인해 주인장 몸상태가 완전히 박살나서($\cdots$) 회고를 다음 날 오후에야 작성하는 점 양해 바랍니다... 보통은 아무리 늦어도 수업 다음 날 오전 10시 이전에도 올릴 수 있도록 노력하는데 오늘은 도저히 몸이 안 따라줍니다.

# 좋았던 점
* 오늘부로 기본적인(혹은 전통적인) 머신러닝 방식은 끝났습니다. 이제 27일차부터는 본격적으로 딥러닝에 들어가게 됩니다. ~~하지만 저는 그 첫 수업을 라이브로 듣지 못하게 되었...~~

# 아쉬웠던 점
* 이론을 자세히 설명해주시려는 의도는 잘 알겠지만, 역시 오늘도 수강생들의 수준을 전혀 고려하지 않은 설명은 너무 아쉬웠습니다. 아무래도 라이브러리를 쓰지 않고 직접 구현하는 것부터 보여주려고 하신 것 같은데, 그러려면 이론적 설명을 하더라도 (일부 수강생들에게 부담으로 작용할 수 있는) 수식은 조금 덜어내고 직관적으로 받아들일 수 있게 설명을 하는 것이 우선이고 그 이후에야 함수를 스크래치 빌드해서 설명할 수 있다고 봅니다. </br> 첨언하자면 시중에 나와있는 교재랑 비교해서 직접 보여주신 코드가 딱히 이해가 더 잘 되는 것도 아니었습니다.

# 배운 점
* 선형회귀(linear regression) 복습 및 추가 진도: 선형 회귀라는 이름답게 경계선을 선형으로 그어서 분류를 합니다. 아, 참고로 이름에 회귀가 붙어있기는 하지만, 후술할 로지스틱 회귀와 마찬가지로 이 모델은 회귀 모델이 아니라 **분류 모델**에 속합니다. 다른 머신러닝 서적들에서도 이 점을 강조하고 있으니 주의해서 접근합시다!
* 경사하강법(gradient descent): 머신러닝과 그 하위 분야인 딥러닝에 있어서 가장 핵심이 되는 기법이 바로 경사하강법입니다. 경사하강법(혹은 줄여서 경사법)은 학습 과정에서 오차를 나타내는 비용 함수(cost function, 또는 손실 함수 loss function)을 최소화하는 방식으로, 함수의 현 위치에서의 기울기가 최소가 되는 곳을 향해 학습시키는 방식입니다. 흔히 아래와 같은 수식으로 나타냅니다. </br> $$x_0=x_0-\eta\frac{\partial f}{\partial x_0}$$</br> 여기서 기호 $\eta$(에타)를 학습률(learning rate)이라고 합니다. 학습률이 너무 낮으면 최적화하는 데 시간이 너무 오래 걸리고, 반대로 지나치게 크면 오차가 작아지기는 커녕 오히려 점점 더 커지는 결과를 초래할 수 있습니다. 따라서 최적의 학습률을 시행착오 등을 통해 알아내는 것이 중요합니다.
* 로지스틱 회귀(logistic regression): 선형 회귀의 확장 버전입니다.

# 앞으로 바라는 점
* 찬찬히 생각해봤는데, 다음 파트인 딥러닝 파트에서 가장 먼저 배우는 것이 인공신경망의 개념인데, 이걸 가르치려면 선형회귀 이전에 퍼셉트론의 기본 개념부터 배우는 것이 필요하다고 봅니다. 애초에 (이름은 여러가지이지만) 다층 퍼셉트론(Multi-Layer Perceptron, MLP)이라는 이름에서 시작하는 분야니까요. 게다가 몸 상태 때문에 필요한 서류를 찾아보다가 슬랙에서 그동안의 커리큘럼 예정 진도를 확인해봤는데, 분명 해당 커리큘럼 상에서는 퍼셉트론과 선형 회귀과 앞에 있었고, 지금 강사님의 진도와는 정반대의 순서로 나아가고 있습니다. 갈수록 커리큘럼이 이해가 가지 않는 상황입니다.
* 다만, 일견 그 의도가 이해가 가는 부분이기도 한 것이 대학교를 가지 않고 고등학교 졸업 후 바로 취업하신 분들도 수강하시는 만큼 오늘 수업에 나왔던 미분 관련 사항들을 최대한 뒤로 빼는 것을 의도했을 수도 있습니다. 하지만, 그 부분을 주로 염두에 두셨다 해도 난이도 조정에 큰 실패를 겪었음에 이견의 여지가 없습니다.
